Phosphor/src/edu/columbia/cs/psl/phosphor/struct/LazyArrayIntTags.java

```diff
@@ -33,4 +34,17 @@ public void setTaints(int tag) {
 	}
 
 	public abstract Object getVal();
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		LazyArrayIntTags that = (LazyArrayIntTags) o;
+		return Objects.equals(this.getVal(), that.getVal());
```

Comment:
Along the lines of below right? Based on re-reading the Phosphor paper and looking at similar code like `edu.columbia.cs.psl.phosphor.runtime.CharacterUtils`

```
	public TaintedBooleanWithIntTag equals$$PHOSPHORTAGGED(int taint, Object o, TaintedBooleanWithIntTag ret) {
		ret.val = this.equals(o);
		ret.taint = taint;
	    return ret;
	}
```

Reply:
Yup, exactly! Unfortunately a lot of the details like this never went into the paper because they were optimizations that we put in after we got it working the first time :/